pkg/k8s/scanner/scanner.go

```diff
@@ -143,3 +172,160 @@ func (s *Scanner) filter(ctx context.Context, r types.Report, artifact *artifact
+					RepoDigests: []string{
+						fmt.Sprintf("%s@%s", name, cDigest),
+					},
+				}, ftypes.Package{})
+
+				if err != nil {
+					return nil, xerrors.Errorf("failed to create PURL: %w", err)
+				}
+				imageComponents = append(imageComponents, &core.Component{
+					PackageURL: &imagePURL,
+					Type:       cdx.ComponentTypeContainer,
+					Name:       name,
+					Version:    cDigest,
+					Properties: map[string]string{
+						cyc.PropertyPkgID:   fmt.Sprintf("%s:%s", name, version),
+						cyc.PropertyPkgType: oci,
+					},
+				})
+			}
+			rootComponent := &core.Component{
+				Name:       comp.Name,
+				Type:       cdx.ComponentTypeApplication,
+				Properties: comp.Properties,
+				Components: imageComponents,
+			}
+			coreComponents = append(coreComponents, rootComponent)
+		case nodeInfo:
+			var nf bom.NodeInfo
+			err := ms.Decode(artifact.RawResource, &nf)
+			if err != nil {
+				return nil, err
+			}
+			coreComponents = append(coreComponents, nodeComponent(nf))
+		default:
+			return nil, fmt.Errorf("resource kind %s is not supported", artifact.Kind)
+		}
+	}
+	rootComponent := &core.Component{
+		Name:       clusterName,
+		Type:       cdx.ComponentTypeContainer,
+		Components: coreComponents,
+	}
+	return rootComponent, nil
+}
+
+func sanitizedVersion(version string) string {
+	return strings.TrimPrefix(version, "v")
+}
+
+func osNameVersion(name string) (string, string) {
+	var buffer bytes.Buffer
+	var v string
+	var err error
+	parts := strings.Split(name, " ")
+	for _, p := range parts {
+		_, err = version.Parse(p)
+		if err != nil {
+			buffer.WriteString(p + " ")
+			continue
+		}
+		v = p
+		break
+	}
+	return strings.ToLower(strings.TrimSpace(buffer.String())), v
+}
+
+func runtimeNameVersion(name string) (string, string) {
+	parts := strings.Split(name, "://")
+	if len(parts) == 2 {
+		return parts[0], parts[1]
+	}
+	return "", ""
+}
+
+func nodeComponent(nf bom.NodeInfo) *core.Component {
+	osName, osVersion := osNameVersion(nf.OsImage)
+	runtimeName, runtimeVersion := runtimeNameVersion(nf.ContainerRuntimeVersion)
+	kubeletVersion := sanitizedVersion(nf.KubeletVersion)
+	return &core.Component{
+		Type:       cdx.ComponentTypeContainer,
+		Name:       nf.NodeName,
+		Properties: nf.Properties,
+		Components: []*core.Component{
+			{
+				Type:    cdx.ComponentTypeOS,
+				Name:    osName,
+				Version: osVersion,
+				Properties: map[string]string{
+					"Class": types.ClassOSPkg,
+					"Type":  osName,
+				},
+			},
+			{
+				Type: cdx.ComponentTypeApplication,
+				Name: nodeCoreComponents,
+				Properties: map[string]string{
+					"Class": types.ClassLangPkg,
+					"Type":  golang,
+				},
+				Components: []*core.Component{
+					{
+						Type:    cdx.ComponentTypeLibrary,
+						Name:    kubelet,
+						Version: kubeletVersion,
+						Properties: map[string]string{
+							cyc.PropertyPkgType: golang,
+						},
+						PackageURL: &purl.PackageURL{
```

Comment:
For PURL, we should use canonical names, `k8s.io/kubelet`.
https://github.com/kubernetes/kubelet/blob/e70f2e547c08910ba3c82c34dbfa0108cb1631eb/go.mod#L3

Reply:
Other components as well, like `k8s.io/apiserver`. You can see the list here.
https://github.com/kubernetes/kubernetes/tree/e56002ab0495de16a5c55463c4591f9c4320705b/staging